src/diffusers/models/attention_processor.py

```diff
@@ -369,7 +369,20 @@ def set_use_memory_efficient_attention_xformers(
                 )
                 processor = XFormersAttnAddedKVProcessor(attention_op=attention_op)
             else:
-                processor = XFormersAttnProcessor(attention_op=attention_op)
+                processor = self.processor
+                if isinstance(self.processor, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0)):
+                    processor = IPAdapterXFormersAttnProcessor(hidden_size=self.processor.hidden_size,
+                        cross_attention_dim=self.processor.cross_attention_dim,
+                        scale=self.processor.scale,
+                        num_tokens=self.processor.num_tokens,
+                        attention_op=attention_op)
+                    processor.load_state_dict(self.processor.state_dict())
+                    if len(self.processor._modules) > 0:
```

Comment:
what does this section of code do?

Reply:
This is passing on the initialization parameters that were already defined during the loading of the ip adapter model to the new xformers attention class. After this I reload the state_dict already loaded in the new object, as already explained in the previous question. Then I just make sure that the weights are in the same device and dtype that were previously, because when reloading the state_dict they are placed in "cpu" and with dtype "float32".
I changed this initial if statement in line 380 to check for existing modules just to avoid unexpected errors.
```python
if hasattr(self.processor, "_modules") and len(self.processor._modules) > 0:
```

Reply:
ok I think we can simplify the code here a little bit because we are inside an `if` statement here so we already know the processor will be either `IPAdapterAttnProcessor`, `IPAdapterAttnProcessor2_0` or `IPAdapterXFormersAttnProcessor` -- in all of these 3 cases it will have a `to_k_ip layer` and `to_v_ip layer`, so maybe we can just get device info from `self.to_k_ip[0].device`

Reply:
Oh yes, I had done it in an agnostic way, because I wasn't sure that there would always be these modules in all the models that might arrive there. I'll change it to your solution.